primitives/arithmetic/src/fixed_point.rs

```diff
@@ -77,14 +77,16 @@ pub trait FixedPointNumber:
 
 	/// Returns the negation.
 	fn negate(self) -> Self {
-		Self::from_inner(-self.into_inner())
+		Self::from_inner(Self::Inner::zero().saturating_sub(self.into_inner()))
```

Comment:
Neg(FixedU128::from(10)) is 0 then? perhaps worth renaming the saturating_neg (if that name makes any sense 😁 ). 

Ot if we have a way to know from `Self::inner` if this type is signed or not, we can just return `self` here if this type is unsigned. 

Yeah this is better IMO.

Reply:
Even easier the `trait Fixed` can just have a `const SIGNED: bool` that gets implemented.

Reply:
I think the better way is to modify the macro to add these methods when signed.

Reply:
At the end I decided to not complicate things and just remove `negate()`. Lmk if you disagree.